### PR TITLE
feat: async token & expose interceptors

### PR DESCRIPTION
**Pull request by suevalov**
* feat: add ability to get accessToken asynchnously 422e8e2
* feat: add ability to set custom interceptors 988328c
* test: convert all tests to typescript & async await b5b5d13

## Goals

* To be able to intercept 403 errors without using proxies (see experience-cma-utils repo)
* To be able to get initial token using promise (that it's how all cma clients are initialized in our main web application)

These changes should allow using `contentful-management` in our main web application and to replace questionable proxy-interceptor with a more logical API.

## Todo

- [x] Unit tests for both new features

### CODE REVIEW

File: src/create-http-client.ts [hunk 1]
```diff
@@ -127,6 +129,20 @@ export default function createHttpClient(
       ...newParams,
     })
   }
+
+  if (config.onBeforeRequest) {
```
Comment (anho) [inline src/create-http-client.ts:133]: Does the order of registering the handlers matter?

**Review by anho — COMMENTED** [review]
Can you briefly outline how that would be used in an app?

File: src/create-http-client.ts (hunk 1 above)
Comment (suevalov) [reply src/create-http-client.ts:133]: Yes, ideally the order shouldn't change. Do you want me to write a unit test for this?

**Comment by suevalov** [comment]
@anho I added "goals" section to the description. Could you please take one more look?

**Review by TimBeyer — APPROVED** [review]

File: src/create-http-client.ts (hunk 1 above)
Comment (anho) [reply src/create-http-client.ts:133]: A comment that explains why the order is important is fine.

Comment (suevalov) [reply src/create-http-client.ts:138]: done

**Review by anho — APPROVED** [review]

**Pull request merged**